### ISSUE TITLE
Comments get removed if a self-running function is directly below imports

### ISSUE THREAD

**Issue by bladeSk** [SeinopSys/eslint-plugin-import-newlines#13]
Consider a basic React app, the code is formatted by Prettier using the config `semi: false, singleQuote: true`.
```JSX
import { createRoot } from 'react-dom/client'
import App from './App'

/**
 * Bootstraps React.
 */
;(function startReact() {
  const root = createRoot(document.getElementById('root'))
  root.render(<App />)
})()
```

When running `eslint --fix` with this plugin, the comment is removed and the semicolon is moved. My configuration is `'import-newlines/enforce': [ 'error', { items: Infinity } ]`, but the bug happens with any settings, as far as I can tell. It happens regardless of using the default JS or TS parser.

```JSX
import { createRoot } from 'react-dom/client'
import App from './App';

(function startReact() {
  const root = createRoot(document.getElementById('root'))
  root.render(<App />)
})()
```

The expected behavior is, of course, to keep the comment and preferably leave the semicolon where it was.

**Comment by SeinopSys** [comment]
I have a sneaking suspicion that because you have a `;` in front of the IIFE the comment is treated by the parser as part of the import statement. As far as I know there is no reason to have that there, you can safely remove it and formatting should work as expected.

**Comment by bladeSk** [comment]
You're right the semicolon doesn't need to be there at all. The thing is, Prettier _always_ adds the semicolon in `semi: false` mode and `eslint --fix` then proceeds to remove the comment. Running Prettier+ESLint is a common use case with an official package that does just that https://github.com/prettier/prettier-eslint .

The actual file has multiple IIFEs, they're unaffected by eslint, except for the first one under the import statements block.

**Comment by SeinopSys** [comment]
I have released version 1.3.2 with a workaround for this specific issue, please update to this version and reopen the issue if the problem persists.

**Comment by bladeSk** [comment]
Excellent work, just tested this and I can confirm it's working correctly now, comments and semicolon formatting being preserved.
